frontend/src/pages/details-page.tsx

```diff
@@ -30,6 +37,24 @@ export default function DetailsPage() {
     }
   }, [post]);
 
+  const category = post.categories[0];
+  useEffect(() => {
```

Comment:
It should use all the categories, 
Basically if there are three categories a b and c , i need to get all the posts which has any of a b or c. and then get only top 3 posts in the mobile view, vertically and may be 4 or 5 in the desktop view